CapstoneProject/Models/MoodDetection.m

```diff
@@ -0,0 +1,82 @@
+//
+//  MoodDetection.m
+//  CapstoneProject
+//
+//  Created by Apoorva Chilukuri on 7/15/22.
+//
+
+#import "MoodDetection.h"
+
+@interface MoodDetection ()
+
+@property (strong, nonatomic) NSMutableArray *userTextWords;
+
+@end
+
+@implementation MoodDetection
+
+- (void) initializer{
+    
+    NSArray *sadKeyWords = @[@"sad", @"down", @"terrible", @"depressed"];
+    NSArray *boredKeyWords = @[@"bored", @"ok", @"tired", @"sleepy"];
+    NSArray *happyKeyWords = @[@"happy", @"great", @"esstatic", @"excited"];
+    NSArray *angryKeyWords = @[@"angry", @"mad", @"upset"];
+    
+    
+    self.stopWordsForMoods = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                                  sadKeyWords,    @"sad",
+                              boredKeyWords,  @"bored",
+                              happyKeyWords, @"happy",
+                              angryKeyWords, @"angry",
+                              nil
+    ];
+    
+    self.moodCount = [NSMutableDictionary dictionaryWithObjectsAndKeys:
+                      [NSNumber numberWithInt:0], @"sad",
+                      [NSNumber numberWithInt:0], @"bored",
+                      [NSNumber numberWithInt:0], @"happy",
+                      [NSNumber numberWithInt:0], @"angry",
+                      nil];
+}
+
+- (NSString *) detectMood:(NSString *) userText{
+    
+    [self initializer];
+    
+    NSArray * userTextWords = [userText componentsSeparatedByString:@" "];
+    
+    // counts how many times a key word/ associated word comes up
+    
+    for(int i = 0; i < userTextWords.count; i++){
```

Comment:
```
for (NSString *userTextWord in userTextWords)
```

Try to use the iterator to avoid getting from an array. Getting from array is dangerous sometimes because out of boundary leads to a crash.